DBConnect.cc

```diff
@@ -54,4 +86,216 @@ void ConnectToMongoDB() {
     mongoc_cleanup();
 }
 
+bool SaveLayoutToDB(const std::string& name, const std::string& json_string) {
+    mongoc_client_t* client;
+    mongoc_database_t* database;
+    mongoc_collection_t* collection;
+    bson_t *command, reply, layout;
+    char* str;
+    char user[16];
+    bson_error_t error;
+    bool result = true;
+
+    initMongoDB(&database, &client, &collection, "layouts");
+
+    str = bson_as_json(&reply, NULL);
+
+    cuserid(user);
+
+    bson_init(&layout);
+    BSON_APPEND_UTF8(&layout, "username", user);
+    BSON_APPEND_UTF8(&layout, "name", name.c_str());
+
+    if (!strcmp(json_string.c_str(), "")) {
+        if (!mongoc_collection_delete_one(collection, &layout, NULL, NULL, &error)) {
+            fmt::format("Delete failed: {}", error.message);
+            result = false;
+        }
+    } else {
+        BSON_APPEND_UTF8(&layout, "json_string", json_string.c_str());
+
+        if (!mongoc_collection_insert_one(collection, &layout, NULL, NULL, &error)) {
+            fmt::format("{}", error.message);
+            result = false;
+        }
+    }
+    bson_destroy(&reply);
+    bson_destroy(command);
+    bson_free(str);
+
+    mongoc_collection_destroy(collection);
+    mongoc_database_destroy(database);
+    mongoc_client_destroy(client);
+    mongoc_cleanup();
+
+    return result;
+}
+
+//
+bool GetLayoutsFromDB(CARTA::RegisterViewerAck* ack_message_ptr) {
+    mongoc_client_t* client;
+    mongoc_database_t* database;
+    mongoc_collection_t* collection;
+    mongoc_cursor_t* cursor;
+    bson_t *insert, *query;
+    const bson_t* doc;
+    char* str;
+    char* uvalue = NULL;
+    bool retval;
```

Comment:
Some of these variables are declared but never used. Not sure if `clang-tidy` will pick these up.

Reply:
The unused `retval`s are gone, but there are still unused `uvalue`s.